contracts/crowdfund/AuctionCrowdfundBase.sol

```diff
@@ -302,4 +302,36 @@ abstract contract AuctionCrowdfundBase is Crowdfund {
             revert InvalidAuctionIdError();
         }
     }
+
+    function _finalize(
+        CrowdfundLifecycle lc,
+        IMarketWrapper market_,
+        uint256 auctionId_,
+        uint96 lastBid_
+    ) internal {
+        // Mark as busy to prevent `burn()`, `bid()`, and `contribute()`
+        // getting called because this will result in a `CrowdfundLifecycle.Busy`.
+        _bidStatus = AuctionCrowdfundStatus.Busy;
+        {
+            // If we've bid before or the CF is not expired,
+            // finalize the auction.
+            if (lastBid_ != 0 || lc == CrowdfundLifecycle.Active) {
+                if (!market_.isFinalized(auctionId_)) {
```

Comment:
There will now be a period where `market_.isFinalized` returns `false`, but the crowdfund itself is "finalized." Could that have any implications elsewhere?

Reply:
I don't think so, although I'll keep this in mind.

Reply:
I don't think so, but I'll keep this in mind.